### ISSUE TITLE
adding migrations fails trying to configure EF for config and operational stores with .net 5.0.7

### ISSUE THREAD

**Issue by leadfollowmove** [DuendeSoftware/IdentityServer#292]
Following the tutorial: https://docs.duendesoftware.com/identityserver/v5/quickstarts/4_ef/

Adding the migrations fails for me with the error: "No DbContext named 'PersistedGrantDbContext' was found."

Checked numerous examples online and the code looks to be correct in the startup.cs but they are all .net3.1 examples. Wondered if I'm doing something obviously wrong or if there is an issue with doing this with .net 5.0.7?

` public class Startup
    {
        public IWebHostEnvironment Environment { get; }
        public IConfiguration Configuration { get; }

        public Startup(IWebHostEnvironment environment, IConfiguration configuration)
        {
            Environment = environment;
            Configuration = configuration;
        }

        public void ConfigureServices(IServiceCollection services)
        {

            string connectionString = Configuration.GetConnectionString("DefaultConnection");
            var migrationsAssembly = typeof(Startup).GetTypeInfo().Assembly.GetName().Name;

            services.AddControllersWithViews();

            var builder = services.AddIdentityServer(options =>
            {
                options.Events.RaiseErrorEvents = true;
                options.Events.RaiseInformationEvents = true;
                options.Events.RaiseFailureEvents = true;
                options.Events.RaiseSuccessEvents = true;
                options.UserInteraction.LoginUrl = "/Account/Login";
                options.UserInteraction.LogoutUrl = "/Account/Logout";
                options.Authentication = new AuthenticationOptions()
                {
                    CookieLifetime = TimeSpan.FromHours(10), // ID server cookie timeout set to 10 hours
                    CookieSlidingExpiration = true
                };

                // Docker Name Resolution name used below.
                options.IssuerUri = "http://sts";

                // see https://identityserver4.readthedocs.io/en/latest/topics/resources.html
                options.EmitStaticAudienceClaim = true;
            })
            .AddConfigurationStore(options =>
            {
                options.ConfigureDbContext = b => b.UseSqlServer(connectionString, sql => sql.MigrationsAssembly(migrationsAssembly));
            })
            .AddOperationalStore(options =>
            {
                options.ConfigureDbContext = b => b.UseSqlServer(connectionString, sql => sql.MigrationsAssembly(migrationsAssembly));
                options.EnableTokenCleanup = true;
            });`

**Comment by leastprivilege** [comment]
I will have a look. thanks!

**Comment by leastprivilege** [comment]
Hi, 

I just tried it and it works fine for me.

I ran into versioning related issues though - make sure that all framework versions point to the latest (including the targets file) - and also update the EF global tool to latest.

**Comment by leadfollowmove** [comment]
Thanks @leastprivilege I'll go back and check carefully. Appreciate the investigation and confirming.

**Comment by leadfollowmove** [comment]
Hi,

Apologies, I realized I never said thanks for the information. I'm currently running ID5 as a POC to see if I can get it working.

One of the areas I'm finding a bit of a challenge is understanding the usage/intent for each column of the database when following the Quickstarts. Is the scheme documented to help here?

Things like:

  *   details around the dbo.ApiResourceSecrets table
  *   the emphasize column under dbo.ApiScopes
  *   nonEditable in the dbo.Apiresources table - seems obvious but are there implications on setting this as non-editable
  *   why some tables enforce recording the created field and some don't.

Basically, every table added from Identity, and Configuration details spawns at least one question or curiosity.

Thanks in advance for any pointers to docs on this.

Cheers,
Adam
________________________________

**Comment by leastprivilege** [comment]
Well - they are really on entity versions of our configuration OM

https://docs.duendesoftware.com/identityserver/v5/reference/models/

**Comment by leadfollowmove** [comment]
Thanks Dominick and appreciate the super-fast reply.

I had seen that link on the docs site, but it seemed to lack detail to the table level and looked aimed more of an intro.

I'll explore and play and see how I go.

Cheers,
Adam
________________________________
